src/main/java/de/retest/recheck/ignore/SearchFilterFiles.java

```diff
@@ -37,6 +37,23 @@
 	private static final List<String> defaultWebFilter = Arrays.asList( "positioning.filter", "style-attributes.filter",
 			"invisible-attributes.filter", "content.filter" );
 
+	private static final Collector<Pair<String, FilterLoader>, ?, Map<String, Filter>> loadSpecificFilter =
+			Collectors.toMap(
+					// Use the file name as key.
+					Pair::getLeft,
+					// Use the loaded filter as value.
+					pair -> {
+						final FilterLoader loader = pair.getRight();
+						try {
+							return loader.load();
+						} catch ( final IOException e ) {
+							log.error( "Could not load filter for name '{}'.", pair.getLeft(), e );
+							return Filter.NEVER_MATCH;
+						}
+					},
+					// Prefer specific over generic filters (due to concat order).
+					( specificFilter, genericFilter ) -> specificFilter );
```

Comment:
Since this is now a separate, class-wide collector, we don't control the stream order anymore. Before we had:

```java
Stream.of( getProjectFilterFiles(), getUserFilterFiles(), getDefaultFilterFiles() )
```

So the filter that came first was indeed more specific. But now one could apply the collector anywhere, like:

```java
Stream.of( getDefaultFilterFiles(), getUserFilterFiles(), getProjectFilterFiles() )
```

Then it would be `(  genericFilter, specificFilter ) -> genericFilter`.

Therefore, I suggest to rename the parameters and/or collector and add a few lines of Javadoc. Maybe name it `fifoCollector` and then have `( first, second ) -> first`?

Reply:
That is a valid point. I pushed an alternative approach (I still have the old changes, if you wish).

Collect the `FilterLoader` in a map which does the loading strategy (i.e. fifo). Then load the filters or the specific filter, if present.